### PR TITLE
[SPARTA-508] add delete checkpoints when updating fragments

### PR DESCRIPTION
**Pull request by acemilio**
**Given** an existing policy which contains some fragments
**When** you modify any of these fragments
**Then** the checkpoint associated with this policy must be deleted

### CODE REVIEW

File: serving-api/src/main/scala/com/stratio/sparta/serving/api/service/http/FragmentHttpService.scala [hunk 1]
```diff
@@ -173,10 +178,17 @@ trait FragmentHttpService extends BaseHttpService {
...
-            val future = supervisor ? new Update(fragment)
-            Await.result(future, timeout.duration) match {
-              case Response(Success(fragment)) => HttpResponse(StatusCodes.OK)
-              case Response(Failure(exception)) => throw exception
+            val policyActor = actors.get(AkkaConstant.PolicyActor).get
+            val fragmentStatusActor = actors.get(AkkaConstant.FragmentActor).get
+            for {
+              updateResponse <- fragmentStatusActor ? FragmentActor.Update(fragment)
+              relatedPolicies <- policyActor ? PolicyActor.FindByFragment(fragment.fragmentType,
+                fragment.id.getOrElse(throw new RuntimeException("No valid policy id")))
```
Comment (anistal) [inline serving-api/src/main/scala/com/stratio/sparta/serving/api/service/http/FragmentHttpService.scala]: This exception should be thrown before the for comprehension

File: serving-api/src/main/scala/com/stratio/sparta/serving/api/service/http/FragmentHttpService.scala [hunk 2]
```diff
@@ -173,10 +178,17 @@ trait FragmentHttpService extends BaseHttpService {
...
+            } yield (updateResponse, relatedPolicies) match {
```
Comment (anistal) [inline serving-api/src/main/scala/com/stratio/sparta/serving/api/service/http/FragmentHttpService.scala]: I think that you do not need to yield updateResponse only relatedPolicies will be used in the match

File: serving-api/src/main/scala/com/stratio/sparta/serving/api/service/http/PolicyHttpService.scala [hunk 3]
```diff
@@ -129,8 +129,9 @@ trait PolicyHttpService extends BaseHttpService with SpartaSerializer {
     path(HttpConstant.PolicyPath / "fragment" / Segment / Segment) { (fragmentType, id) =>
       get {
         complete {
-          val future = supervisor ? new FindByFragment(fragmentType, id)
-          Await.result(future, timeout.duration) match {
+          for{
+            response <- supervisor ? FindByFragment(fragmentType, id)
+          }yield response match {
```
Comment (anistal) [inline serving-api/src/main/scala/com/stratio/sparta/serving/api/service/http/PolicyHttpService.scala]: correct indentation and spaces here:
for{
...
}yield

to

for {
...
} yield

File: serving-api/src/test/scala/com/stratio/sparta/serving/api/service/http/HttpServiceBaseTest.scala [hunk 4]
```diff
@@ -26,13 +27,13 @@ import com.stratio.sparta.serving.core.models._
 import com.stratio.sparta.serving.core.policy.status.PolicyStatusEnum
 
 /**
```
Comment (anistal) [inline serving-api/src/test/scala/com/stratio/sparta/serving/api/service/http/HttpServiceBaseTest.scala]: This indentation is wrong. Please fix your editor with our standard.

File: serving-api/src/test/scala/com/stratio/sparta/serving/api/utils/PolicyUtilsTest.scala [hunk 5]
```diff
@@ -1,33 +1,38 @@
 /**
```
Comment (anistal) [inline serving-api/src/test/scala/com/stratio/sparta/serving/api/utils/PolicyUtilsTest.scala]: Fix your editor

File: serving-api/src/main/scala/com/stratio/sparta/serving/api/service/http/FragmentHttpService.scala (hunk 2 above)
Comment (acemilio) [inline serving-api/src/main/scala/com/stratio/sparta/serving/api/service/http/FragmentHttpService.scala]: I disagree. If updateResponse fails and returns a Failure, you need to include in pattern matching, with relatedPolicies.

**Comment by acemilio** [comment]
Added some fixes and line indentions (fixed my editor). 

I disagree with reference to yield, or not, updateResponse. If updateResponse fails and returns a Failure, you need to include in pattern matching, with relatedPolicies.

**Comment by anistal** [comment]
+1

**Comment by danielcsant** [comment]
+1

**Comment by antonioalf** [comment]
:shipit:

**Comment by stratiocommit** [comment]
Comment accepted. Will treat it ASAP

**Comment by stratiocommit** [comment]
Squashing and checks in progress.

**Pull request merged**